app/objects/score.py

```diff
@@ -282,30 +289,58 @@ async def from_submission(cls, data: list[str]) -> Score:
         # 15 osu_version + (" " * client_flags)
         """
 
-        s.online_checksum = data[0]
-
-        s.n300, s.n100, s.n50, s.ngeki, s.nkatu, s.nmiss, s.score, s.max_combo = map(
-            int,
-            data[1:9],
-        )
-
+        s.client_checksum = data[0]
+        s.n300 = int(data[1])
+        s.n100 = int(data[2])
+        s.n50 = int(data[3])
+        s.ngeki = int(data[4])
+        s.nkatu = int(data[5])
+        s.nmiss = int(data[6])
+        s.score = int(data[7])
+        s.max_combo = int(data[8])
         s.perfect = data[9] == "True"
-        _grade = data[10]  # letter grade
+        s.grade = Grade.from_str(data[10])
         s.mods = Mods(int(data[11]))
         s.passed = data[12] == "True"
         s.mode = GameMode.from_params(int(data[13]), s.mods)
-
-        # TODO: we might want to use data[14] to get more
-        #       accurate submission time (client side) but
-        #       we'd probably want to check if it's close.
-        s.play_time = datetime.now()
-
+        s.client_time = datetime.strptime(data[14], "%y%m%d%H%M%S")
         s.client_flags = ClientFlags(data[15].count(" ") & ~4)
 
-        s.grade = Grade.from_str(_grade) if s.passed else Grade.F
+        s.server_time = datetime.now()
 
         return s
 
+    def compute_online_checksum(
```

Comment:
should be cached_property, `server_checksum`, to remain consistent with `client_checksum`